BedrockCore.cpp

```diff
@@ -268,6 +332,77 @@ BedrockCore::RESULT BedrockCore::processCommand(unique_ptr<BedrockCommand>& comm
     return needsCommit ? RESULT::NEEDS_COMMIT : RESULT::NO_COMMIT_REQUIRED;
 }
 
+void BedrockCore::postProcessCommand(unique_ptr<BedrockCommand>& command) {
+    AutoTimer timer(command, BedrockCommand::POSTPROCESS);
+    BedrockServer::ScopedStateSnapshot snapshot(_server);
+    command->lastPeekedOrProcessedInState = _server.getState();
+
+    // Convenience references to commonly used properties.
+    const SData& request = command->request;
+    SData& response = command->response;
+    STable& content = command->jsonContent;
+
+    // We catch any exception and handle in `_handleCommandException`.
+    try {
+        SDEBUG("postProcessing at '" << request.methodLine << "' with priority: " << command->priority);
+        uint64_t timeout = _getRemainingTime(command, false);
+        command->postProcessCount++;
+
+        _db.startTiming(timeout);
+
+        if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
+            STHROW("501 Failed to begin shared postProcess transaction");
+        }
+
+        // Make sure no writes happen while in postProcess command
+        _db.setQueryOnly(true);
+
+        // postProcess.
+        command->reset(BedrockCommand::STAGE::POSTPROCESS);
+        command->postProcess(_db);
+        SDEBUG("Plugin '" << command->getName() << "' postProcess command '" << request.methodLine << "'");
+
+        // If no response was set, assume 200 OK
+        if (response.methodLine == "") {
```

Comment:
Will this ever be true if `process` does this same block?

Reply:
We have the same thing in `peek` though, too. So I'm a bit confused why we would ever have it is `process` if we have it in `peek`.